source/slang/slang-emit-cpp.cpp

```diff
@@ -1724,6 +1724,10 @@ void CPPSourceEmitter::emitRTTIObject(IRRTTIObject* rttiObject)
 void CPPSourceEmitter::_maybeEmitWitnessTableTypeDefinition(
     IRInterfaceType* interfaceType)
 {
+    if (m_interfaceTypesEmitted.Contains(interfaceType))
+        return;
+    m_interfaceTypesEmitted.Add(interfaceType);
+
```

Comment:
It's a bit unfortunate to have to add this logic. Why can't we just emit the witness table type when we encounter the actual instruction that represents the type?

Reply:
Yeah, I think it is just me being lazy to refactor the code that way.